src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -207,25 +216,42 @@ protected override void WriteStartChangesetImplementation(string changeSetId)
         /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
         /// <param name="payloadUriOption">
         /// The format of operation Request-URI, which could be AbsoluteUri, AbsoluteResourcePathAndHost, or RelativeResourcePath.</param>
-        /// <param name="dependsOnIds">The prerequisite request ids of this request.</param>
+        /// <param name="dependsOnIds">The prerequisite request ids of this request. It should be null for Multipart/Mixed format
+        /// and, therefore, validation will pass automatically.</param>
         /// <returns>The message that can be used to write the request operation.</returns>
         protected override ODataBatchOperationRequestMessage CreateOperationRequestMessageImplementation(
             string method, Uri uri, string contentId, BatchPayloadUriOption payloadUriOption,
             IEnumerable<string> dependsOnIds)
         {
+            if (dependsOnIds != null)
+            {
+                throw new ODataException(
+                    "For Multipart/Mixed batch format, request dependency is implicit by the protocol and should be null.");
+            }
+
             // write pending message data (headers, response line) for a previously unclosed message/request

```

Comment:
Do not enforce null. We should validate that any ids specified are in the ids read by the tracker for this scope (changeset or toplevel).

Reply:
removed null check / exception thrown

---
In reply to: [158350729](https://github.com/OData/odata.net/pull/1020#discussion_r158350729) [](ancestors = 158350729)